src/Abp/Runtime/Caching/CacheExtensions.cs

```diff
@@ -91,6 +92,35 @@ public static Task<object[]> GetAsync(this ICache cache, string[] keys, Func<Tas
             return Task.WhenAll(tasks);
         }
 
+        public static bool TryGetValue<TKey, TValue>(this ICache cache, TKey key, out TValue value)
+        {
+            return cache.TryGetValue(key.ToString(), out value);
+        }
+
+        public static async Task<ConditionalValue<TValue>> TryGetValueAsync<TKey, TValue>(this ICache cache, TKey key)
+        {
+            var result = await cache.TryGetValueAsync(key.ToString());
+            return new ConditionalValue<TValue>(result.HasValue, result.Value == null ? default : (TValue)result.Value);
```

Comment:
Why not `return cache.TryGetValueAsync(key.ToString());`?

Reply:
cache is having the interface IAbpCache<string, object>, the result of ``cache.TryGetValueAsync`  will be `ConditionalValue<object>`, we need to cast the object into TValue

Reply:
I don't think these should be extensions (since casting can fail), but rather internal implementations in `TypedCacheWrapper` (which seems to be the place that these are used).

Reply:
@acjh i can do the clean up in a follow up PR

Reply:
No, we shouldn't add these extensions in this PR.

Reply:
I made a clean up PR for these extensions at https://github.com/aspnetboilerplate/aspnetboilerplate/pull/5394, after that is merge, i will move these methods from extension into `TypeCacheWrapper`